### PR TITLE
[DOR-105] Page timed out error page

### PR DESCRIPTION
**Pull request by rhague-hmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/traderservices/views/templates/GovukLayoutWrapper.scala.html [hunk 1]
```diff
@@ -51,6 +51,15 @@
...
 </noscript>
 }
+
+  <!-- TODO this <meta> should only be included when the user is logged in -->
+  <!-- TODO set correct timeout (900), countdown (780) and keep-alive-url -->
+  <meta name="hmrc-timeout-dialog" content="hmrc-timeout-dialog"
+  data-language="en"
+  data-timeout="@appConfig.timeout"
+  data-countdown="@appConfig.countdown"
+  data-keep-alive-url="/trader-services/keep-alive"
```
Comment (arturopala) [inline app/uk/gov/hmrc/traderservices/views/templates/GovukLayoutWrapper.scala.html:61]: these URLs shall not be hardcoded, pls inject Routes and reference the routes

File: app/uk/gov/hmrc/traderservices/wiring/AppConfig.scala [hunk 2]
```diff
@@ -107,6 +111,10 @@ class AppConfigImpl @Inject() (config: ServicesConfig) extends AppConfig {
   override val signOutUrl: String = config.getString("urls.signOut")
   override val researchBannerUrl: String = config.getString("urls.researchBanner")
 
+  override val timeout: Int = config.getInt("session._timeoutSeconds")
```
Comment (arturopala) [inline app/uk/gov/hmrc/traderservices/wiring/AppConfig.scala:114]: why underscore?

File: it/uk/gov/hmrc/traderservices/stubs/DataStreamStubs.scala [hunk 3]
```diff
@@ -49,6 +49,20 @@ trait DataStreamStubs extends Eventually {
     stubFor(post(urlPathEqualTo(auditUrl + "/merged")).willReturn(aResponse().withStatus(204)))
   }
 
+  def givenTimedOut(): Unit =
```
Comment (arturopala) [inline it/uk/gov/hmrc/traderservices/stubs/DataStreamStubs.scala:52]: This stub is for DataStream calls only, pls use other file

File: it/uk/gov/hmrc/traderservices/support/BaseISpec.scala [hunk 4]
```diff
@@ -35,6 +35,8 @@ abstract class BaseISpec
   override def commonStubs(): Unit = {
     givenCleanMetricRegistry()
     givenAuditConnector()
+    givenTimedOut()
```
Comment (arturopala) [inline it/uk/gov/hmrc/traderservices/support/BaseISpec.scala:38]: there is no need to add it here as this is not called during every test

Comment (arturopala) [reply it/uk/gov/hmrc/traderservices/support/BaseISpec.scala:38]: pls move it to the tests

File: it/uk/gov/hmrc/traderservices/support/TestAppConfig.scala [hunk 5]
```diff
@@ -28,4 +28,7 @@ case class TestAppConfig(wireMockBaseUrl: String, wireMockPort: Int) extends App
   override val subscriptionJourneyUrl: String = "/subscription"
 
   val fileFormats: AppConfig.FileFormats = AppConfig.FileFormats(10, "", "")
+  override val timeout: Int = 10
+  override val countdown: Int = 2
+  override val timedoutUrl: String = "http://localhost:9379/trader-services/timedout"
```
Comment (arturopala) [inline it/uk/gov/hmrc/traderservices/support/TestAppConfig.scala:33]: this is not correct, test server can run on a random port

File: app/uk/gov/hmrc/traderservices/views/templates/GovukLayoutWrapper.scala.html (hunk 1 above)
Comment (slashrealm) [reply app/uk/gov/hmrc/traderservices/views/templates/GovukLayoutWrapper.scala.html:61]: Sure thing, it's a placeholder URL for now, as it's still WIP

File: app/uk/gov/hmrc/traderservices/wiring/AppConfig.scala (hunk 2 above)
Comment (rhague-hmrc) [reply app/uk/gov/hmrc/traderservices/wiring/AppConfig.scala:116]: I copied it from another hmrc project so I thought it was there for a reason

File: conf/application.conf [hunk 6]
```diff
@@ -51,6 +51,12 @@ controllers {
   }
 }
 
+session{
+  timeoutSeconds = 900
+  countdownInSeconds = 120
+  timeoutUrl = "http://localhost:9379/trader-services/timedout"
```
Comment (arturopala) [inline conf/application.conf:57]: `timeoutUrl` is our own URL and there is no need to configure it via AppConfig, pls use `routes.SessionController.showTimeoutPage` directly in the layout wrappers

File: app/uk/gov/hmrc/traderservices/wiring/AppConfig.scala [hunk 7]
```diff
@@ -63,15 +63,20 @@ trait AppConfig {
...
-  val researchBannerUrl: String
 
+  val researchBannerUrl: String
   val fileFormats: AppConfig.FileFormats
 
   val traceFSM: Boolean = false
+
+  val timeout: Int
+  val countdown: Int
+  val timedoutUrl: String
```
Comment (arturopala) [inline app/uk/gov/hmrc/traderservices/wiring/AppConfig.scala:79]: we don't need this to be configured, the `timedout` URL is stable and can be referenced using 
 `appConfig.baseExternalCallbackUrl + routes.SessionController.showTimeoutPage.url`

File: app/uk/gov/hmrc/traderservices/views/templates/GovukLayoutWrapper2.scala.html [hunk 8]
```diff
@@ -58,6 +59,15 @@
...
 </noscript>
 }
+
+@if(timeout) {
+  <meta name="hmrc-timeout-dialog" content="hmrc-timeout-dialog"
+  data-language="en"
+  data-timeout="@appConfig.timeout"
+  data-countdown="@appConfig.countdown"
+  data-keep-alive-url="@routes.SessionController.keepAlive"
+  data-sign-out-url="@appConfig.timedoutUrl"/>
```
Comment (arturopala) [inline app/uk/gov/hmrc/traderservices/views/templates/GovukLayoutWrapper2.scala.html:69]: pls use `@{appConfig.baseExternalCallbackUrl + routes.SessionController.showTimeoutPage.url}`

**Pull request merged**